src/Nethermind/Nethermind.JsonRpc/Modules/Parity/ParityRpcModule.cs

```diff
@@ -67,11 +67,16 @@ public class ParityRpcModule : IParityRpcModule
             _peerManager = peerManager ?? throw new ArgumentNullException(nameof(peerManager));
         }
 
-        public ResultWrapper<ParityTransaction[]> parity_pendingTransactions()
-            => ResultWrapper<ParityTransaction[]>.Success(_txPool.GetPendingTransactions().Where(pt => pt.SenderAddress != null)
+        public ResultWrapper<ParityTransaction[]> parity_pendingTransactions(Address? address = null)
+        {
+           IEnumerable<Transaction> enumerable = address == null
+                ? _txPool.GetPendingTransactions().Where(pt => pt.SenderAddress != null)
+                : _txPool.GetPendingTransactionsBySender(address);
+            return ResultWrapper<ParityTransaction[]>.Success(enumerable
                 .Select(t => new ParityTransaction(t, Rlp.Encode(t).Bytes,
-                    t.IsSigned ? _ecdsa.RecoverPublicKey(t.Signature, t.Hash) : null)).ToArray());
-        
+                t.IsSigned ? _ecdsa.RecoverPublicKey(t.Signature, t.Hash) : null)).ToArray());
+        }
```

Comment:
Hm... btw @marcindsobczak do we need this special code to recover public key, and filter null sender address? I think those kind of transactions won't get to tx pool in the first place?

Reply:
Yeah, `null` sender shouldn't be accepted in `TxPool`. I just wanted this to be a quick, one-line additional feature, without changing old code, not sure if it's used often enough to be worth more effort. But right, I think we can remove here `Where` with `null` check. Not sure about `PublicKey`, it's needed in constructor of `ParityTransaction` and is a part of OE tx response (https://openethereum.github.io/JSONRPC.html#transaction-responses)

Reply:
so there is this filter : UnknownSenderFilter for the sender filtering so i guess we can remove the null sender check.
but the RecoverPublicKey is needed since you need to provide the public key with the ParityTransaction. @LukaszRozmej  let me know what you would like to do,